extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIProcessor.java

```diff
@@ -663,11 +664,38 @@ private GACT getGACT(String artifactKey) {
     }
 
     private Class toClass(Type type) {
-        try {
-            return tccl.loadClass(type.name().toString());
-        } catch (ClassNotFoundException ex) {
-            throw new RuntimeException(ex);
+        if (type.kind().equals(Type.Kind.PRIMITIVE)) {
+            return toPrimitiveClass(type.asPrimitiveType());
+        } else if (type.kind().equals(Type.Kind.VOID)) {
+            throw new RuntimeException("Void method return detected, JsonRPC Method needs to return something.");
+        } else {
+            try {
+                return tccl.loadClass(type.name().toString());
+            } catch (ClassNotFoundException ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+    }
+
+    private Class toPrimitiveClass(PrimitiveType primitiveType) {
+        if (primitiveType.equals(PrimitiveType.BOOLEAN)) {
+            return boolean.class;
+        } else if (primitiveType.equals(PrimitiveType.BYTE)) {
+            return byte.class;
+        } else if (primitiveType.equals(PrimitiveType.CHAR)) {
+            return char.class;
+        } else if (primitiveType.equals(PrimitiveType.DOUBLE)) {
+            return double.class;
+        } else if (primitiveType.equals(PrimitiveType.FLOAT)) {
+            return float.class;
+        } else if (primitiveType.equals(PrimitiveType.INT)) {
+            return int.class;
+        } else if (primitiveType.equals(PrimitiveType.LONG)) {
+            return long.class;
+        } else if (primitiveType.equals(PrimitiveType.SHORT)) {
+            return short.class;
         }
+        throw new RuntimeException(primitiveType.primitive() + " is not a supported return type");
```

Comment:
@Ladicek don't we have something in Jandex nowadays?

Reply:
Yeah, `JandexReflection.loadRawType()` should work here.

Reply:
Thanks @Ladicek . I did not know about that. Works like a charm :)